python/automatic-memory-pinning.yml

```diff
@@ -0,0 +1,8 @@
+rules:
+- id: automatic-memory-pinning
+  patterns:
+    - pattern: torch.utils.data.DataLoader(...)
+    - pattern-not-inside: torch.utils.data.DataLoader(..., pin_memory=True, ...)
```

Comment:
1) will it work with `x = torch.utils.data; x.DataLoader(...)`?
2) I guess it will miss cases with `kwargs = {'pin_memory': True}; torch.utils.data.DataLoader(**kwargs)`
3) I guess it will miss cases with `torch.utils.data.DataLoader(pin_memory=1)` - does this work in torch?

Is there any way we can catch 2) and 3)? If not, we should document it.

Reply:
I updated the rule; it now works with all 3 of those cases.

Reply:
It also now works with any non-zero integer.

Reply:
I wasn't able to capture all the cases with the original rule. More concretely, I wasn't able to write a rule that would capture instances when `pin_memory` isn't set at all and when it is set to `False`. As I was doing more research into this issue, it turns out there are a lot of cases where automatic memory pinning is not possible, so I decided to flag cases where the argument isn't set. That way, the rule will find instances where the implications of automatically pinning memory were not considered. Let me know if you think I should revert this rule.